src/main/java/org/reactome/release/downloaddirectory/MapOldStableIds.java

```diff
@@ -79,50 +83,107 @@ public static void execute(String username, String password, String host, String
 			{
 				deferredIds.add(stableIdsArray);
 			} else {
-				preferredIds.add(stableIdsArray);
+				stableIdsToOldIdsMappings.add(stableIdsArray);
 			}
 		}
-		preferredIds.addAll(deferredIds);
+		stableIdsToOldIdsMappings.addAll(deferredIds);
+
+		logger.info("Retrieving current stable identifiers from " + dba.getDBName());
+		Set<String> currentStableIdentifiers = getCurrentStableIdentifiers(dba);
 
 		// Write to file
-		File oldIdMappingFile = new File(filename);
-		if (oldIdMappingFile.exists()) {
-			oldIdMappingFile.delete();
-		}
-		oldIdMappingFile.createNewFile();
-		String header = "# Reactome stable IDs for release " + releaseNumber + "\n" + "Stable_ID\told_identifier(s)\n";
-		Files.write(Paths.get(filename), header.getBytes(), StandardOpenOption.APPEND);
-		for (List<Object> stableIdsArray : preferredIds)
-		{
-			String primaryId = (String) stableIdsArray.get(0);
-			@SuppressWarnings("unchecked")
-			List<String> secondaryIds = (ArrayList<String>) stableIdsArray.get(1);
-			String line = primaryId + "\t" + String.join(",", secondaryIds) + "\n";
-			Files.write(Paths.get(filename), line.getBytes(), StandardOpenOption.APPEND);
-		}
-		String outpathName = releaseNumber + "/" + filename;
-		Files.move(Paths.get(filename), Paths.get(outpathName), StandardCopyOption.REPLACE_EXISTING);
+		Files.deleteIfExists(Paths.get(filename));
+		Files.createFile(Paths.get(filename));
```

Comment:
I think there's a way to create/overwrite-if-exists instead of two separate statements.

Reply:
I think this would work: 
```Java
Files.write(path, content.getBytes(), StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING );
```
(not here, but in the place that you actually want to write the file content)